pytorch_lightning/metrics/metric.py

```diff
@@ -332,8 +332,23 @@ def _filter_kwargs(self, **kwargs):
     def __hash__(self):
         hash_vals = [self.__class__.__name__]
 
+        # Torch tensors are hashable, but based on their
+        # underlying pointer. Since we do a structural
+        # equality check in __eq__, we circumvent this
+        # by hashing the _sum_ of tensors' contents
+        def format_tensor(t):
```

Comment:
Please, no local functions, make it a staticmethod
add the above comments as a docstring to the function.
```suggestion
        def hash_tensor(t):
```

Reply:
Let's make this function private `_hash_tensor`